pkg/cloudprovider/controller/nodecontroller_test.go

```diff
@@ -552,7 +552,7 @@ func TestSyncCloudDeletePods(t *testing.T) {
 	}
 }
 
-func TestHealthCheckNode(t *testing.T) {
+func TestNodeConditionsCheck(t *testing.T) {
```

Comment:
As this is expanding quickly, can you add some comments on each case:
1. test setup
2. expectation

We shouldn't test by method, we need to test by behavior, i.e. this method probably needs to be splitted to multiple ones.  I'm fine with it right now.